### ISSUE TITLE
Web based version of nin10kit

### ISSUE THREAD

**Issue by TricksterGuy** [TricksterGuy/nin10kit#6]

**Comment by Shubhang** [comment]
Would this require a total rewrite of the program?

**Comment by TricksterGuy** [comment]
No.  Only a frontend interface would need to be written, almost all of the code can be reused.

One could write a webpage using a form asking for images, and options mirroring that of the CLI interface, and when submitted, the server could just form the proper command and execute the cli version of the program server side and then display the output or have the user download it in some way.

Another approach would be to simply make bindings for most of the internal library stuff (stuff like SWIG, CLIF, Boost.Python) for a more web friendly programming language (ruby for Ruby on Rails, and I'm sure python has a web framework),  Then you can write a website as normally and then you can just write the web page in whatever language you want, and then call the bound methods which could then run the C++ code internally.

And there may be some C++ libraries (like emscripten) that would make it easy to port a desktop application for the web instead.
